### PR TITLE
detect/analyzer: add more details for the tcp window keyword-v1

### PR DESCRIPTION
**Pull request by Nancyenos**
Ticket: 6352

Make sure these boxes are checked accordingly before submitting your Pull Request -- thank you.

## Contribution style:
- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html

## Our Contribution agreements:
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)

Link to ticket: https://redmine.openinfosecfoundation.org/issues/6352

Describe changes:
-detect/analyzer: add more details for the tcp window keyword
-I need help with this, am  I in the right direction?
-Also am thinking of below tests and would like to know if i need to add more, i tested for exact and negated size
```
alert tcp any any -> any any (msg:"TCP window size"; window:30336; sid:1;)
alert tcp any any -> any any (msg:"TCP window size"; window:!1024; sid:2;)
```

=

### CODE REVIEW

File: src/detect-engine-analyzer.c [hunk 1]
```diff
@@ -932,6 +933,14 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
                 jb_close(js);
                 break;
             }
+            case DETECT_WINDOW: {
+                const DetectWindowData *wd = (const DetectWindowData *)smd->ctx;
+                jb_open_object(js, "window");
+                jb_set_uint(js, "size", wd->size);
+                jb_set_bool(js, "negate", wd->negated);
```
Comment (jufajardini) [inline src/detect-engine-analyzer.c:940]:
```suggestion
                jb_set_bool(js, "negated", wd->negated);
```

**Review by jufajardini — CHANGES_REQUESTED** [review]
This looks like the right approach to me, too :)

One minor change needed (a typo), please see inline comment.

The rules you've shared for the test candidates look good to me, maybe we could have one of them use `tcp.window`, instead of `window`,  just to showcase that both work.

**Comment by Nancyenos** [comment]
Thankyou so much for the help @jufajardini

**Comment by jufajardini** [comment]
Work continued with https://github.com/OISF/suricata/pull/12024

**Pull request closed without merging**